source/particle/property/crystal_preferred_orientation.cc

```diff
@@ -677,10 +682,65 @@ namespace aspect
 
         // create shortcuts
         const std::array<double, 4> &tau = ref_resolved_shear_stress;
-
         std::vector<double> strain_energy(n_grains);
         double mean_strain_energy = 0;
 
+        // CPX and olivine_d_fabric has different slip systems
+        // first initiate the l and n for olivine A,B,C,E types
```

Comment:
`l` and `n` is not clear for someone not familiar with CPO. Change the comment to the proper names.